### PR TITLE
Added Gesture Recognizer to MapView for ORKLocationAnswerFormat.

### PR DESCRIPTION
**Pull request by slichlyter12**
Press and hold on map for one second to place marker and input location into text field.

It currently jumps to the nearest address, which is reasonable for my purposes, but others may want a more refined placement.

At the moment there is no animation happening when the user places a marker, this I think would a good addition but not entirely sure where to start with that.

### CODE REVIEW

**Comment by slichlyter12** [comment]
Here's a GIF of the current implementation:

orklocationanswerformat-gesturerecognizer

File: ResearchKit/Common/ORKLocationSelectionView.m [hunk 1]
```diff
@@ -153,6 +154,27 @@ - (instancetype)initWithFormMode:(BOOL)formMode
...
 
+- (void)setUpGestureRecognizer {
+    UILongPressGestureRecognizer *lpgr = [[UILongPressGestureRecognizer alloc] initWithTarget:self action:@selector(addPlacemarkToMap:)];
+    lpgr.minimumPressDuration = 1.0; // press for 1 second
+    [_mapView addGestureRecognizer:lpgr];
+}
+
+- (void)addPlacemarkToMap:(UIGestureRecognizer *)gestureRecognizer {
+    if (gestureRecognizer.state != UIGestureRecognizerStateBegan)
+        return;
```
Comment (YuanZhu-apple) [inline ResearchKit/Common/ORKLocationSelectionView.m]: add `{}` around it.

File: ResearchKit/Common/ORKLocationSelectionView.m [hunk 2]
```diff
@@ -153,6 +154,27 @@ - (instancetype)initWithFormMode:(BOOL)formMode
...
+    
+    CGPoint touchPoint = [gestureRecognizer locationInView:_mapView];
+    CLLocationCoordinate2D touchMapCoordinate = [_mapView convertPoint:touchPoint toCoordinateFromView:_mapView];
+    
+    MKPointAnnotation *annotation = [[MKPointAnnotation alloc] init];
```
Comment (YuanZhu-apple) [inline ResearchKit/Common/ORKLocationSelectionView.m]: Prefer `new` over `alloc and init`

**Comment by YuanZhu-apple** [comment]
@slichlyter12 Can you redo this PR base on current master?

**Comment by slichlyter12** [comment]
@YuanZhu-apple I created a new pull request: #807

**Pull request closed without merging**